catalog/rocks/catalog.go

```diff
@@ -73,10 +73,6 @@ type Entry struct {
 	ETag         string
 }
 
-func (e *Entry) IsTombstone() bool {
```

Comment:
Are we going to have different types `Entry` in mutable and immutable?  Because, while the immutables interface never needs a tombstone, AFAICT the mutables interface _must_ have one if we want to layer mutables on top of immutables.  (The only alternative I can see is to have mutables know about immutables and overlay for itself… which just ends up making a different internal "entry" type there.)